testfixtures/logcapture.py

```diff
@@ -174,11 +248,13 @@ def check(self, *expected):
           A sequence of entries of the structure specified by the ``attributes``
           passed to the constructor.
         """
-        return compare(
+        result = compare(
             expected,
             actual=self.actual(),
             recursive=self.recursive_check
             )
+        self.mark_logs_as_asserted()
+        return result
```

Comment:
I don't think this should mark as checked. If you're relying on `str` to do checks (which is pretty much a deprecated behaviour now anyway) and you have checking for assertions turned on, then you probably want to have a manual `mark_all_checked ` call to be explicit about your intentions.

Hopefully you have a test for this and you can change it to ensure that str'ing does *not* ensure checked.

Reply:
I think you need to expand this diff, the mark as checked in not part of the str method, it is part of the check method.

Reply:
perfect!